pipelines/services/validate.go

```diff
@@ -0,0 +1,116 @@
+package services
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/gookit/color"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+const (
+	validatePipResourcePath = "api/v1/validateYaml"
+)
+
+type ValidateService struct {
+	client *jfroghttpclient.JfrogHttpClient
+	auth.ServiceDetails
+}
+
+func NewValidateService(client *jfroghttpclient.JfrogHttpClient) *ValidateService {
+	return &ValidateService{client: client}
+}
+
+func (vs *ValidateService) getHttpDetails() httputils.HttpClientDetails {
+	httpDetails := vs.ServiceDetails.CreateHttpClientDetails()
+	return httpDetails
+}
+
+func (vs *ValidateService) ValidatePipeline(data []byte) error {
+	var err error
+	httpDetails := vs.getHttpDetails()
+
+	// Query params
+	m := make(map[string]string, 0)
+
+	// URL Construction
+	uri := vs.constructValidateAPIURL(m, validatePipResourcePath)
+
+	// Headers
+	headers := make(map[string]string)
+	headers["Content-Type"] = "application/json"
+	httpDetails.Headers = headers
+	log.Debug(string(data))
+
+	// Send post request
+	resp, body, err := vs.client.SendPost(uri, data, &httpDetails)
+	if err != nil {
+		return err
+	}
+	if err := errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return err
+	}
+
+	// Process response
+	if nil != resp && resp.StatusCode == http.StatusOK {
+		log.Info("Processing resources yaml response ")
+		log.Info(string(body))
+		err = processValidatePipResourceResponse(body)
+		if err != nil {
+			log.Error("Failed to process resources validation response")
+		}
+	}
```

Comment:
You already check the status code in CheckResponseStatusWithBody. Please also consider nil before "CheckResponseStatusWithBody".